base/datafmt.jl

```diff
@@ -38,16 +38,16 @@ function countlines(io::IO, eol::Char)
     nl
 end
 
-readdlm(input, T::Type; opts...) = readdlm(input, invalid_dlm, T, '\n'; opts...)
-readdlm(input, dlm::Char, T::Type; opts...) = readdlm(input, dlm, T, '\n'; opts...)
+readdlm{T}(input, ::Type{T}; opts...) = readdlm(input, invalid_dlm, T, '\n'; opts...)
```

Comment:
Is it treated differently upon inclining? I was under the (perhaps misguided) impression that this was also the better style in general. I'll re-examine this PR with more consideration.